---
title: "Quantification"
author: "Chi Lam Poon"
date: "3/21/2021"
output: html_document
---


## Running featureCounts

```bash
spack load -r subread@1.6.2
ls
SRR12953537.sorted.bam  SRR12953550.sorted.bam  SRR12953565.sorted.bam  SRR12953576.sorted.bam
SRR12953546.sorted.bam  SRR12953555.sorted.bam  SRR12953571.sorted.bam  SRR12953586.sorted.bam
SRR12953548.sorted.bam  SRR12953556.sorted.bam  SRR12953575.sorted.bam  SRR12953588.sorted.bam

# gene level
files=$(ls *.bam)
featureCounts -T 62 -t exon -g gene_id -a ../../refGenome/hg38.ensGene.gtf.gz -o all.counts.txt $files
```

### Looking into quantification situations
```{r quant}
library(tidyr)
library(ggplot2)

summ <- read.delim("/athena/angsd/scratch/zhp4001/data/alignments/all.counts.txt.summary")
colnames(summ) <- gsub(".sorted.bam", "", colnames(summ))
summ <- summ[sapply(summ$Status, function(x) sum(summ[summ$Status==x,-1]) !=0), ]
summ <- summ %>% gather(sample, count, -Status)
ggplot(summ, aes(x=sample, y=count, fill=Status)) + 
  geom_bar(stat="identity", position="dodge") + 
  theme_bw() +
  coord_flip() + 
  theme(legend.position = "bottom")
```
Why there are so many `Unassigned_NoFeatures`...? The overall alignment rates of HISAT2 were all ~94,95%, fastqc & bamqc reports also look good (one samples' were shown in a previous homework and the others not shown here).

Maybe there are too many reads falled on intronic regions and other non-coding regions..?
I tried to view the reads on IGV using one sample and they looked fine...

![](/athena/angsd/scratch/zhp4001/data/fastqc/IGV.png)

Load the read counts:

```{r reads}
cnts <- read.delim("/athena/angsd/scratch/zhp4001/data/alignments/all.counts.txt", comment.char = "#")
rownames(cnts) <- cnts[,1]
cnts <- cnts[,-c(1:6)]
colnames(cnts) <- gsub(".sorted.bam", "", colnames(cnts))

nrow(cnts)

getColors <- function(vec) {
  # assgin a color to each group of samples
  library(RColorBrewer)
  n <- length(unique(vec))
  col <- brewer.pal(n, "Paired")
  #col <- brewer.pal.info[brewer.pal.info$category=='qual', ] # get max. 74 colours
  #col_all <- unlist(mapply(brewer.pal, col$maxcolors, rownames(col)))
  ifelse (n > length(col), 
          cvec <- sample(col, n, replace=T),
          cvec <- sample(col, n, replace=F)
  )
  vec <- as.character(vec)
  names(vec) <- rep(NA, length(vec))
  for (g in 1:length(unique(vec))) {
    names(vec)[which(vec==unique(vec)[g])] <- cvec[g]
  }
  vec
}

dens.plot <- function(table, colors, ...) {
  cols <- names(colors)
  d <- plot(density(table[, 1]), col=cols[1], 
            lwd=2, las=2, xlab="", ...) +
    abline(v=0, lty=3) + title(xlab="log2 exprs", ylab=NA) +
    for (i in 2:ncol(table)) {
      den <- density(table[, i])
      lines(den$x, den$y, col=cols[i], lwd=2)
    } 
  legend('topright', legend=unique(colors), lty=1, col=unique(names(colors)), cex=0.6)
  d
}

colors <- getColors(colnames(cnts))
dens.plot(log2(cnts+1), colors, main="Expression distribution")
```

The distribution looks good.


Import into DESeq2
```{r de}
suppressPackageStartupMessages({
  suppressWarnings({
  library(DESeq2)
  library(magrittr)
  library(pheatmap)
})
})

# make information file
samples.info <- read.delim("/athena/angsd/scratch/zhp4001/ANGSD-project/data/SraRunTable.txt", sep = ",")
samples.info <- samples.info[, c("Run", "AGE", "gender", "pmi", "Race", "Tissue")]
# manually add one more variable
samples.info$TOD <- c(1.688509416, 11.7744848, 17.09167991, 8.419554181, -1.167548126, -5.330048689, 
                      -0.619766727, 14.58263798, 6.385902741, 15.86508723, 7.58795633, 3.639673108)
samples.info


# generate deseq object
DESeq.ds <- DESeqDataSetFromMatrix(countData = cnts , colData = samples.info, design = ~ AGE + gender + pmi + TOD)
```

Note: PMI is postmortem interval.

```{r de-2, fig.width=8}
# remove genes without any counts
DESeq.ds <- DESeq.ds[ rowSums(counts(DESeq.ds)) > 0, ]
# different library sizes
colSums(counts(DESeq.ds))

# calculate the size factor and add it to the data set
DESeq.ds <- estimateSizeFactors(DESeq.ds)
colData(DESeq.ds)

norm.cnts <- counts(DESeq.ds, normalized = TRUE)

# log transformation
assay(DESeq.ds, "logCounts") <- log2(norm.cnts + 1)

# rlog transformation
DESeq.rlog <- rlog(DESeq.ds, blind = TRUE)
assay(DESeq.ds, "rlog") <- assay(DESeq.rlog)

corr_coeff <- cor(assay(DESeq.ds, "rlog"), method = "pearson")
as.dist(1-corr_coeff, upper = TRUE) %>% as.matrix %>% pheatmap(., main = "Pearson correlation", 
                                                               labels_row = colData(DESeq.ds)$gender,
                                                               labels_col = colData(DESeq.ds)$gender)
```
Seems hard to differentiate female & male samples for now, I think PCA will get an ambiguous result like this. Let's see if any DE genes in the next step.

### Saving
```{r save}
saveRDS(DESeq.ds, file="/athena/angsd/scratch/zhp4001/ANGSD-project/data/deseq.obj.rds")
```

```{r session}
sessionInfo()
```
