---
title: "Final Report"
author: "Chi Lam Poon"
date: "4/18/2021"
output: 
  rmarkdown::html_document:
    toc: true
    toc_float:
      collapsed: false
    highlight: tango
    df_print: paged
---

# Sex differences of transcriptomes in normal human putamen tissues

## Introduction
Sex differences in brains in terms of neurodevelopment, neurochemistry and susceptibility to neurodegenerative and neuropsychiatric disease have been reported. The underlying molecular basis of these differences remains unclear and it is worth exploring ([Trabzuni et al., 2013](https://www.nature.com/articles/ncomms3771)).

Diurnal rhythms in gene expression have been detected in the human brain. Evidence also suggests sex differences in physiological and behavioral rhythms, for example, women tend to have earlier wake and sleep times and earlier melatonin onset, etc. ([Logan et al, 2021](https://www.sciencedirect.com/science/article/abs/pii/S0006322321011100))

A recently published study ([Ketchesin et al 2021](https://www.pnas.org/content/118/2/e2016150118)) identified diurnal rhythms in transcript expression across three striatal subregions (caudate, putamen, and nucleus accumbens), with postmortem brain tissues from subjects without psychiatric or neurological disorders. Using time-of-death (TOD) analysis, they found that core circadian clock genes are rhythmic across all three regions and show strong phase concordance across regions, but the putamen contains a much larger number of significantly rhythmic transcripts than the other two regions.

Here I would like to use their putamen datasets to test whether there are transcripts with differential expression between 6 females' and males' putamens. Confounders mentioned in the metadata (race, time of death, etc.) will be corrected. Regarding the sample size, it's 6 female and 6 male for now.

Hypotheses:

1. The linear model in differential expression analysis is not easy to detect some core circadian genes (ARNTL, PER, CRY), i.e. these genes would not show differential expression when the coefficient is time of death (TOD);
2. There are indeed differentially expressed genes identified in putamens between female & male.


## Results

```{r load, echo=F}
suppressWarnings({
  suppressPackageStartupMessages({
  library(DESeq2)
  library(dplyr)
  library(ggplot2)
  library(edgeR)
  library(ggbeeswarm)
  library(ggpubr)
})
})
suppressMessages(dds <- readRDS("/athena/angsd/scratch/zhp4001/ANGSD-project/data/deseq.obj.rds"))
```

### DE - TOD
#### Results
```{r design1}
dds@design
dds <- DESeq(dds)
anno <- metadata(dds)[["annotation"]]
# try to get DEGs for TOD
res.tod <- results(dds)
res.tod
# take a look
resOrdered.tod <- res.tod[order(res.tod$padj),] %>% data.frame()
resOrdered.tod$geneName <- anno[match(rownames(resOrdered.tod), anno$id),]$symbol
head(resOrdered.tod)
DESeq2::plotMA(res.tod, ylim=c(-2,2))
```
Only two genes have < 0.05 adjusted P-values. Also, no core circadian genes ranked high. Using DE analysis (DESeq) to find cycling may not be a good idea because the relationship between TOD and expression is not linear but cosinar/sinosodal.

### DE - gender

#### Results
```{r design2}
# try to get DEGs for sex
res.sex <- results(dds, contrast = c("gender","male","female"))
res.sex
resOrdered.sex <- res.sex[order(res.sex$padj),] %>% data.frame()
resOrdered.sex$geneName <- anno[match(rownames(resOrdered.sex), anno$id),]$symbol
head(resOrdered.sex)

DESeq2::plotMA(res.sex, ylim=c(-5,5))
```

Also try the DESeq2 shrinkage:
```{r shrinkage}
library(apeglm)
resultsNames(dds)
resLFC <- lfcShrink(dds, coef="gender_male_vs_female", type="apeglm")
resLFC
DESeq2::plotMA(resLFC, ylim=c(-5,5))
```

#### Manual volcano plot:
```{r vol}
resOrdered.sex$color <- ifelse(abs(resOrdered.sex$log2FoldChange)>=log2(2) & resOrdered.sex$padj <= 0.05, "steelblue", "lightgrey")

ggplot(resOrdered.sex, aes(x=log2FoldChange, y=-log10(padj), color=color)) +
  geom_point(size=1.5) + theme_bw() + scale_color_identity() + 
  geom_vline(xintercept=c(-log2(2), log2(2)), alpha=0.6, linetype="dotted") +
  geom_hline(yintercept=-log10(0.05), alpha=0.8, color="steelblue", linetype="dotted") +
  ggtitle("Male vs Female") +
  annotate("text", x=-10, y=230, label="XIST", fontface = 'italic') +
  annotate("text", x=12, y=110, label="DDX3Y", fontface = 'italic') +
  annotate("text", x=16, y=80, label="USP9Y", fontface = 'italic') +
  annotate("text", x=-3, y=40, label="TSIX", fontface = 'italic')
```

#### Swarm plot
```{r splot}
cpms <- cpm(assay(dds))
log.cpms <- log2(cpms) + 1

plotting <- function(mat, gene, name) {
  meta <- colData(dds) %>% data.frame()
  gene.cpm <- log.cpms[rownames(log.cpms)==gene,]
  gene.cpm[gene.cpm==-Inf] <- 0
  df <- data.frame(obj=names(gene.cpm), cpm=gene.cpm, gender=meta$gender)
  
  ggplot(df, aes(gender, cpm, color=gender)) + geom_quasirandom() + theme_bw() +
    ggtitle(name) + ylab("log2(cpm)")
}

p1 <- plotting(log.cpms, "ENSG00000229807.12", "XIST")
p2 <- plotting(log.cpms, "ENSG00000270641.1", "TSIX")
p3 <- plotting(log.cpms, "ENSG00000067048.17", "DDX3Y")
p4 <- plotting(log.cpms, "ENSG00000114374.13", "USP9Y")
ggarrange(p1,p2,p3,p4,ncol=2, nrow=2,legend = "right", common.legend = T)
```
Some (not many) reads in female samples were incorrectly mapped to Y genes

#### Gene biotypes
```{r biotype}
resOrdered.sex$biotype <- anno[match(rownames(resOrdered.sex), anno$id),]$biotype
types <- resOrdered.sex[resOrdered.sex$color=="steelblue",]$biotype
table(types)
```

#### Chromosomes for genes
```{r chrom}
resOrdered.sex$chrom <- anno[match(rownames(resOrdered.sex), anno$id),]$chrom
chroms <- resOrdered.sex[resOrdered.sex$color=="steelblue",]$chrom
tmp <- table(chroms)
sum(tmp[!names(tmp) %in% c("chrX","chrY")])
chr.df <- data.frame(chr=c("Autosomal", "X", "Y"), count=c(1648,41,27))
ggplot(chr.df, aes(x=chr,y=count,fill=chr)) +
  geom_bar(position="stack", stat="identity") + theme_bw()
```

The majority is still autosomal genes, but those highly differentiated are genes on sexual chromosomes.

## Methods
### Overview

- Alignment: HISAT2
- Quality Control: FastQC, RSeQC, BamQC, multiQC
- Quantification: featureCounts
- Differential expression analysis: DESeq2

### Getting Data
Based on this publication [Diurnal rhythms across the human dorsal and ventral striatum](https://www.pnas.org/content/118/2/e2016150118), the brain tissue sampels can be found on GEO [GSE160521](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE160521) with sra [table](https://www.ncbi.nlm.nih.gov/Traces/study/?acc=PRJNA673437&o=acc_s%3Aa).

Brain tissue samples were obtained through the University of Pittsburgh Brain Tissue Donation Program and the NIH NeuroBioBank, the lab at UPitt generated the data.

Total RNA was extracted from the tissue samples using a combination of Trizol (Invitrogen, Carlsbad, CA) and RNeasy Lipid Tissue Mini Kit (Qiagen, Hilden, Germany). RNA quantity and quality were assessed using fluorometry (Qubit RNA Broad Range Assay Kit and Fluorometer; Invitrogen, Carlsbad, CA) and chromatography (Bioanalyzer and RNA 6000 Nano Kit; Agilent, Santa Clara, CA), respectively. 

Libraries were prepped for RNA-sequencing using the TruSeq Stranded Total RNA Sample Preparation Kit (Illumina, San Diego, CA). 75bp paired-end dual-indexed sequencing was performed using the __NextSeq 500__ platform (Illumina, San Diego, CA). 

I first selected 6 male & 6 female putamen samples and downloaded the `SraRunTable`.
```{r read}
runTable <- read.csv("/athena/angsd/scratch/zhp4001/ANGSD-project/data/SraRunTable.csv")
runTable
```

Then download samples using sratoolkit:
```bash
cd /home/zhp4001/proj.angsd/data
DIR=/athena/angsd/scratch/zhp4001/sra/sra # in the default sratoolkit directory I set
SRR=$(cat SraRunTable.txt | cut -d, -f1 | sed '1d')
for s in $SRR; do
prefetch $s
fasterq-dump --split-files $DIR/$s.sra -O /athena/angsd/scratch/zhp4001/data
done
```

### QC
```bash
spack load fastqc
fastqc SRR12953537_1.fastq.gz SRR12953537_2.fastq.gz
# forgot to create a directory
mkdir -p fastqc/SRR12953537
mv *fastqc.* fastqc/SRR12953537
```

The Sequences flagged as poor quality of both fqs are 0, and from the qc reports seems like the reads are of high quality. Let's try trimgalore then compare with the reports after trimming.

```bash
spack load -r trimgalore
trim_galore --illumina --stringency 5 -e 0.15 --fastqc --paired SRR12953537_1.fastq.gz SRR12953537_2.fastq.gz
#fastqc SRR12953537_1_trimmed.fq.gz SRR12953537_2_trimmed.fq.gz
```

Looks like I __don't__ need to trim the adapters since values in the `Adapter Content` plot are all ~0. 

And fastqc report after trimming looks similar. There is a peak in this `Per sequence GC content` plot in both untrimmed & trimmed fqs:

![peak](/athena/angsd/scratch/zhp4001/data/fastqc/peak.png)

This is probably caused by the highly repeated reads:
![repeats](/athena/angsd/scratch/zhp4001/data/fastqc/repeats.png)

### Alignment

[HISAT2](http://daehwankimlab.github.io/hisat2/) is a fast and sensitive alignment program. They designed and implemented a graph FM index (GFM), an original approach and its first implementation.

##### Buiding index
```bash
cd /athena/angsd/scratch/zhp4001/refGenome
# reference genome
wget ftp://ftp.ensembl.org/pub/release-84/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
gunzip Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
mv Homo_sapiens.GRCh38.dna.primary_assembly.fa genome.fa

# Build HFM index
hisat2-build -p 16 genome.fa genome
```

##### Aligning
```bash
cd /athena/angsd/scratch/zhp4001/data
mkdir alignments

for s in $(ls *.fastq.gz| sed 's/_.*.gz//g' | sort -u); do
echo Aligning $s...
hisat2 -q -x ../refGenome/genomeIdx/genome -1 ${s}_1.fastq.gz -2 ${s}_2.fastq.gz -S alignments/$s.sam
# convert sam to bam
samtools view -bS alignments/$s.sam | samtools sort > alignments/$s.sorted.bam
rm alignments/$s.sam
done
```

Paramters: 

- `-q`: inputing fastq files
- `-x`: the basename of the index for the reference genome, it's 'genome' here
- `-1`: paired-end seq file 1
- `-2`: paired-end seq file 2
- `-S`: file to write SAM alignments to, maybe S for 'SAM'

```
Aligning SRR12953537...
50817747 reads; of these:
  50817747 (100.00%) were paired; of these:
    6182910 (12.17%) aligned concordantly 0 times
    42503692 (83.64%) aligned concordantly exactly 1 time
    2131145 (4.19%) aligned concordantly >1 times
    ----
    6182910 pairs aligned concordantly 0 times; of these:
      639055 (10.34%) aligned discordantly 1 time
    ----
    5543855 pairs aligned 0 times concordantly or discordantly; of these:
      11087710 mates make up the pairs; of these:
        5863044 (52.88%) aligned 0 times
        4518721 (40.75%) aligned exactly 1 time
        705945 (6.37%) aligned >1 times
94.23% overall alignment rate
```

All samples have ~94%, 95% alignment rates and they are good.

### Counting

#### Running featureCounts

```bash
spack load -r subread@1.6.2
ls
SRR12953537.sorted.bam  SRR12953550.sorted.bam  SRR12953565.sorted.bam  SRR12953576.sorted.bam
SRR12953546.sorted.bam  SRR12953555.sorted.bam  SRR12953571.sorted.bam  SRR12953586.sorted.bam
SRR12953548.sorted.bam  SRR12953556.sorted.bam  SRR12953575.sorted.bam  SRR12953588.sorted.bam

# gene level
files=$(ls *.bam)
featureCounts -T 62 -t exon -g gene_id -a ../../refGenome/gencode.v37.annotation.gtf.gz -o all.counts.2.txt $files
```

Note: The library prep used the TruSeq Stranded kit, which means the reads are stranded. The default behaviour for `-s` flag is 0, i.e. count on both strand, and using `-s 0` should be fine. If sets to `-s 1`, the alignment rates will be much lower. 

#### Looking into quantification situations
```{r quant}
library(tidyr)
library(ggplot2)

summ <- read.delim("/athena/angsd/scratch/zhp4001/data/alignments/all.counts.2.txt.summary")
colnames(summ) <- gsub(".sorted.bam", "", colnames(summ))
summ <- summ[sapply(summ$Status, function(x) sum(summ[summ$Status==x,-1]) !=0), ]
summ <- summ %>% gather(sample, count, -Status)
ggplot(summ, aes(x=sample, y=count, fill=Status)) + 
  geom_bar(stat="identity", position="dodge") + 
  theme_bw() +
  coord_flip() + 
  theme(legend.position = "bottom")
```

#### Pathway analysis
I threw the DEGs into gProfiler to see if any pathway significantly enriched. Since only 12 DEGs in the negative direction, I put up/downregulated genes together to the gProfiler server. Then got this plot:

![](/athena/angsd/scratch/zhp4001/ANGSD-project/img/gprofiler.png)

No pahtway is significant... Pre-ranked GSEA with t-statistics could be better to find differential pathways.

### Differential expression analysis

Load the read counts:
```{r reads}
cnts <- read.delim("/athena/angsd/scratch/zhp4001/data/alignments/all.counts.2.txt", comment.char = "#")
rownames(cnts) <- cnts[,1]
cnts <- cnts[,-c(1:6)]
colnames(cnts) <- gsub(".sorted.bam", "", colnames(cnts))

nrow(cnts)

getColors <- function(vec) {
  # assgin a color to each group of samples
  library(RColorBrewer)
  n <- length(unique(vec))
  col <- brewer.pal(n, "Paired")
  #col <- brewer.pal.info[brewer.pal.info$category=='qual', ] # get max. 74 colours
  #col_all <- unlist(mapply(brewer.pal, col$maxcolors, rownames(col)))
  ifelse (n > length(col), 
          cvec <- sample(col, n, replace=T),
          cvec <- sample(col, n, replace=F)
  )
  vec <- as.character(vec)
  names(vec) <- rep(NA, length(vec))
  for (g in 1:length(unique(vec))) {
    names(vec)[which(vec==unique(vec)[g])] <- cvec[g]
  }
  vec
}

dens.plot <- function(table, colors, ...) {
  cols <- names(colors)
  d <- plot(density(table[, 1]), col=cols[1], 
            lwd=2, las=2, xlab="", ...) +
    abline(v=0, lty=3) + title(xlab="log2 exprs", ylab=NA) +
    for (i in 2:ncol(table)) {
      den <- density(table[, i])
      lines(den$x, den$y, col=cols[i], lwd=2)
    } 
  legend('topright', legend=unique(colors), lty=1, col=unique(names(colors)), cex=0.6)
  d
}

colors <- getColors(colnames(cnts))
dens.plot(log2(cnts+1), colors, main="Expression distribution")
```
The distribution looks good.

Import into DESeq2
```{r de}
suppressPackageStartupMessages({
  suppressWarnings({
  library(DESeq2)
  library(magrittr)
  library(pheatmap)
})
})

# make information file
samples.info <- read.csv("/athena/angsd/scratch/zhp4001/ANGSD-project/data/SraRunTable.csv", sep = ",")
samples.info <- samples.info[, c("Run", "AGE", "gender", "pmi", "Race", "Tissue")]
# manually add one more variable
samples.info$TOD <- c(1.688509416, 11.7744848, 17.09167991, 8.419554181, -1.167548126, -5.330048689, 
                      -0.619766727, 14.58263798, 6.385902741, 15.86508723, 7.58795633, 3.639673108)
samples.info


# generate deseq object
DESeq.ds <- DESeqDataSetFromMatrix(countData = cnts , colData = samples.info, design = ~ AGE + gender + pmi + TOD)
```

Adding annotation file:
```{r anno}
# add annotation
anno <- read.delim("/athena/angsd/scratch/zhp4001/refGenome/annotation.txt", header = F, 
                   col.names = c("id", "chrom", "start","end","symbol","biotype"))
```

DESeq2:
```{r de-2, fig.width=8}
# GENE FILTERS
dim(DESeq.ds)
## removes genes without any counts
DESeq.ds <- DESeq.ds[rowSums(counts(DESeq.ds)) > 0, ]

# removes Y genes
# anno <- anno[!anno$chrom %in% c("chrY"),] 
# DESeq.ds <- DESeq.ds[rownames(DESeq.ds) %in% anno$id, ]

## discards genes expressed in fewer than half of the subjects
library(edgeR)
minSamples <- floor(ncol(DESeq.ds) / 2)
cpms <- cpm(assay(DESeq.ds))
keep <- rowSums(cpms > 2) >= minSamples
sum(keep) #how many genes left
DESeq.ds <- DESeq.ds[keep,]
dim(DESeq.ds)

## add annotation file
anno <- anno[match(rownames(DESeq.ds), anno$id),]
all(as.character(anno$id) == as.character(rownames(DESeq.ds)))
metadata(DESeq.ds)[["annotation"]] <- anno

# different library sizes
colSums(counts(DESeq.ds))

# calculate the size factor and add it to the data set
DESeq.ds <- estimateSizeFactors(DESeq.ds)
colData(DESeq.ds)

norm.cnts <- counts(DESeq.ds, normalized = TRUE)

# log transformation
assay(DESeq.ds, "logCounts") <- log2(norm.cnts + 1)

# rlog transformation
DESeq.rlog <- rlog(DESeq.ds, blind = TRUE)
assay(DESeq.ds, "rlog") <- assay(DESeq.rlog)

corr_coeff <- cor(assay(DESeq.ds, "rlog"), method = "pearson")
as.dist(1-corr_coeff, upper = TRUE) %>% as.matrix %>% pheatmap(., main = "Pearson correlation", 
                                                               labels_row = colData(DESeq.ds)$gender,
                                                               labels_col = colData(DESeq.ds)$gender)
```

PCA:
```{r pca}
vsd <- vst(DESeq.ds, blind=FALSE)
plotPCA(vsd, intgroup=c("gender"))
#saveRDS(DESeq.ds, file="/athena/angsd/scratch/zhp4001/ANGSD-project/data/deseq.obj.rds")
```


## Discussion

### Issues

1. The sample size (6v6) is relatively small
2. The peak in the `Per sequence GC content` plot in both untrimmed & trimmed fqs in some of the samples:
![](/athena/angsd/scratch/zhp4001/ANGSD-project/img/fastqc_per_sequence_gc_content_plot.png)

I tried to use BLAT in UCSC genome browser to see the locations of the first five reads and got:

```
r1       75     1    75    75   100.0%  chr9                  -     9442160   9442234     75
r1       75     1    75    75   100.0%  chr3                  -    15738615  15738689     75
r1       75     1    75    75   100.0%  chr14                 -    49586680  49586754     75
r1       75     1    75    75   100.0%  chr14                 +    49862675  49862749     75
=========
r2       75     1    75    75   100.0%  chr9                  -     9442165   9442239     75
r2       75     1    75    75   100.0%  chr3                  -    15738620  15738694     75
r2       75     1    75    75   100.0%  chr14                 -    49586685  49586759     75
r2       75     1    75    75   100.0%  chr14                 +    49862670  49862744     75
=========
r3       75     1    75    75   100.0%  chr9                  -     9442164   9442238     75
r3       75     1    75    75   100.0%  chr3                  -    15738619  15738693     75
r3       75     1    75    75   100.0%  chr14                 -    49586684  49586758     75
r3       75     1    75    75   100.0%  chr14                 +    49862671  49862745     75

========
r4       75     1    75    75   100.0%  chr9                  -     9442191   9442265     75
r4       75     1    75    75   100.0%  chr14                 -    49586711  49586785     75
r4       75     1    75    75   100.0%  chr14                 +    49853709  49853783     75
r4       75     1    75    75   100.0%  chr14                 +    49862644  49862718     75
========
r5       75     1    75    75   100.0%  chr9                  -     9442163   9442237     75
r5       75     1    75    75   100.0%  chr3                  -    15738618  15738692     75
r5       75     1    75    75   100.0%  chr14                 -    49586683  49586757     75
r5       75     1    75    75   100.0%  chr14                 +    49862672  49862746     75
```

Looks like they are all mapped to similar regions in chr9,3,14, which are probalbly low-complexity sequences. Also, the GC% of these repeated reads are:

```python
# in python
>>> def GC(str):
...   return (str.count('G') + str.count('C')) / len(str)
... 
>>> GC('CCTTAGGCAACCTGGTGGTCCCCCGCTCCCGGGAGGTCACCATATTGATGCCGAACTTAGTGCGGACACCCGATC')
0.6133333333333333
>>> GC('CCCCTCCTTAGGCAACCTGGTGGTCCCCCGCTCCCGGGAGGTCACCATATTGATGCCGAACTTAGTGCGGACACC')
0.6266666666666667
>>> GC('CCCTCCTTAGGCAACCTGGTGGTCCCCCGCTCCCGGGAGGTCACCATATTGATGCCGAACTTAGTGCGGACACCC')
0.6266666666666667
>>> GC('CTCCGTTTCCGACCTGGGCCGGTTCACCCCTCCTTAGGCAACCTGGTGGTCCCCCGCTCCCGGGAGGTCACCATA')
0.6666666666666666
>>> GC('CCTCCTTAGGCAACCTGGTGGTCCCCCGCTCCCGGGAGGTCACCATATTGATGCCGAACTTAGTGCGGACACCCG')
0.6266666666666667
```
Quite matched with the value of the peak... I think it's OK to not to remove them in downstream analyses.


3. There are so many `Unassigned_NoFeatures`... 
fastqc & bamqc reports also look good (one samples' were shown in a previous homework and the others not shown here).

Updated: I've tried `hg38.ensGene.gtf.gz`, `gencode.v37.annotation.gtf.gz`, `gencode.v37.chr_patch_hapl_scaff.annotation.gtf.gz`. The `Unassigned_NoFeature` are quite high in all cases.

Maybe there are too many reads falled on intronic regions and other non-coding regions,
I tried to view the reads on IGV using one sample and they looked fine...

![](/athena/angsd/scratch/zhp4001/data/fastqc/IGV.png)

But the RSeQC result indeed shows many reads on intron:

```
Total Reads                   90985118
Total Tags                    100770957
Total Assigned Tags           0
=====================================================================
Group               Total_bases         Tag_count           Tags/Kb             
CDS_Exons           38789783            0                   0.00              
5'UTR_Exons         23662943            0                   0.00              
3'UTR_Exons         55740806            0                   0.00              
Introns             1568978817          0                   0.00              
TSS_up_1kb          25968254            0                   0.00              
TSS_up_5kb          116328599           0                   0.00              
TSS_up_10kb         211957672           0                   0.00              
TES_down_1kb        28077014            0                   0.00              
TES_down_5kb        121868284           0                   0.00              
TES_down_10kb       217117660           0                   0.00              
=====================================================================
```

A hypothesis to explain the prevalence is that they harbour functional non-coding RNA ([Kapranov et al. 2011](http://www.biomedcentral.com/1741-7007/8/149)).

> By RNA mass in a human cell, transcripts emanating from intronic sequences approximately equal that of exonic sequences but this large amount of intronic sequence cannot be explained just by the fact that introns are longer and, thus, accumulate more reads. The density of reads from individual introns can be quite abundant and similar to, or higher than, that of exonic regions. This is exemplified by the known ncRNA KCNQ1OT1 embedded within the protein-coding KCNQ1 locus and transcribed from the opposite strand, indicating it is not simply a splicing artifact (Figure 3). Additional examples in loci not currently known to harbour ncRNAs are shown on Figure 4b.

Another paper ([Ameur et al, 2011](https://www.nature.com/articles/nsmb.2143)) shows that part of the intronic alignments reflect nascent transcription and co-transcriptional splicing. 

### About the results

It's more interesting to identify (differentially) cycling genes using curve fitting or so with this kind of datasets, as well as to perform pathway analysis using those genes. See the following plot:

![](/athena/angsd/scratch/zhp4001/ANGSD-project/img/paper_res.png)

DE analysis cannot reveal much insights. Also it'll be quite interesting to identify oscillations in single cell data, but it may be pretty hard to generate scRNA-seq with the time-of-death information; or to find oscillating usages of TSS/PolyA/splicing sites.


Session info:
```{r session}
sessionInfo()
```
